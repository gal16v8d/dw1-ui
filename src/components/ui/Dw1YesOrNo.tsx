import { Badge } from 'primereact/badge';

interface Dw1YesOrNoProps {
  value: boolean;
}

const Dw1YesOrNo: React.FC<Dw1YesOrNoProps> = ({ value }): JSX.Element => {
  return (
    <Badge
      value={value ? 'yes' : 'no'}
      severity={value ? 'success' : 'error'}
      className="p-mr-2"
    ></Badge>
  );
};

export default Dw1YesOrNo;
