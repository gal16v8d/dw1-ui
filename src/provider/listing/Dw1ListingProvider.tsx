import { TFunction } from 'i18next';
import { Messages } from 'primereact/messages';
import React from 'react';
import { useTranslation } from 'react-i18next';

interface Dw1ListingContextProps {
  t: TFunction<'translation', undefined>;
  message: React.RefObject<Messages>;
}

const Dw1ListingContext = React.createContext<
  Dw1ListingContextProps | undefined
>(undefined);

const Dw1ListingProvider = (
  props: React.PropsWithChildren<Record<string, unknown>>
): JSX.Element => {
  const { t } = useTranslation();
  const message = React.useRef<Messages>(null);

  return (
    <Dw1ListingContext.Provider value={{ t, message }}>
      {props.children}
    </Dw1ListingContext.Provider>
  );
};

function useListingContext(): Dw1ListingContextProps {
  const context = React.useContext(Dw1ListingContext);
  if (context === undefined) {
    throw new Error('useListingContext must be used within Dw1ListingProvider');
  }
  return context;
}

export { Dw1ListingProvider, useListingContext };
