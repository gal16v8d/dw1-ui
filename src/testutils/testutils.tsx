import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

import { render as rltRender } from '@testing-library/react';
import i18n from 'config/i18n/languageConfig';
import { I18nextProvider } from 'react-i18next';

const renderWithBaseProviders = (ui: ReactElement) => {
  return rltRender(
    <QueryClientProvider client={new QueryClient()}>
      <I18nextProvider i18n={i18n}>{ui}</I18nextProvider>
    </QueryClientProvider>
  );
};

export * from '@testing-library/react';
export { renderWithBaseProviders };
