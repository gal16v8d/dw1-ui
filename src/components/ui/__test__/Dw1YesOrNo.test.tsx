import { renderWithBaseProviders, screen } from 'testutils/testutils';
import { Dw1ListingProvider } from 'provider/listing/Dw1ListingProvider';
import Dw1YesOrNo from '../Dw1YesOrNo';

it('should render Badge component and match snapshot', () => {
  const element = renderWithBaseProviders(
    <Dw1ListingProvider>
      <Dw1YesOrNo value />
    </Dw1ListingProvider>
  );
  expect(element.baseElement).toMatchSnapshot();
});

it.each`
  value    | expectedText
  ${true}  | ${'yes'}
  ${false} | ${'no'}
`('should render Badge component and find text', ({ value, expectedText }) => {
  renderWithBaseProviders(
    <Dw1ListingProvider>
      <Dw1YesOrNo value={value} />
    </Dw1ListingProvider>
  );
  const badgeText = screen.getByText(expectedText);
  expect(badgeText).toBeInTheDocument();
});
