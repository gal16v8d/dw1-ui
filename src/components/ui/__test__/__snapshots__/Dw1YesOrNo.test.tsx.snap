// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Badge component and match snapshot 1`] = `
<body>
  <div>
    <span
      class="p-badge p-component p-badge-success p-mr-2"
    >
      yes
    </span>
  </div>
</body>
`;
